plugins/aws/provisioner.go

```diff
@@ -2,42 +2,155 @@ package aws
 
 import (
 	"context"
+	"fmt"
+	"os"
 
 	"github.com/1Password/shell-plugins/sdk"
 	"github.com/1Password/shell-plugins/sdk/provision"
 	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+	confighelpers "github.com/99designs/aws-vault/v7/vault"
 )
 
 type awsProvisioner struct {
-	stsProvisioner    STSProvisioner
-	envVarProvisioner provision.EnvVarProvisioner
+	selectedProvisioner sdk.Provisioner
 }
 
 func AWSProvisioner() sdk.Provisioner {
-	return awsProvisioner{
-		envVarProvisioner: provision.EnvVarProvisioner{
-			Schema: defaultEnvVarMapping,
-		},
-	}
+	return awsProvisioner{}
 }
 
 func (p awsProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput, out *sdk.ProvisionOutput) {
-	totp, hasTotp := in.ItemFields[fieldname.OneTimePassword]
-	mfaSerial, hasMFASerial := in.ItemFields[fieldname.MFASerial]
-
-	if hasTotp && hasMFASerial {
-		p.stsProvisioner.MFASerial = mfaSerial
-		p.stsProvisioner.TOTPCode = totp
-		p.stsProvisioner.Provision(ctx, in, out)
-	} else {
-		p.envVarProvisioner.Provision(ctx, in, out)
+	provisioner := p.selectCorrespondingProvisioner(in.ItemFields, out)
+	if len(out.Diagnostics.Errors) > 0 {
+		return
 	}
+
+	provisioner.Provision(ctx, in, out)
 }
 
 func (p awsProvisioner) Deprovision(ctx context.Context, in sdk.DeprovisionInput, out *sdk.DeprovisionOutput) {
-	// Nothing to do here: environment variables get wiped automatically when the process exits.
+	p.selectedProvisioner.Deprovision(ctx, in, out)
 }
 
 func (p awsProvisioner) Description() string {
-	return p.envVarProvisioner.Description()
+	if p.selectedProvisioner != nil {
+		return p.selectedProvisioner.Description()
+	}
+	return "Provisions AWS with temporary credentials."
```

Comment:
I am not sure where the `Description` function is supposed to be called. If this function is meant to be called always before `Provision` this line is redundant because `Provision` is the one that sets the `selectedProvisioner`

Reply:
not applicable anymore